src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationTracker.cs

```diff
@@ -145,18 +145,6 @@ public bool ContainsAssemblyOrModuleOrDynamic(ISymbol symbol, bool primary)
                     return new CompilationTracker(newProject, newState);
                 }
 
-                var declarationOnlyCompilation = state.DeclarationOnlyCompilation;
-                if (declarationOnlyCompilation != null)
-                {
-                    if (translate != null)
-                    {
-                        var intermediateProjects = ImmutableArray.Create((this.ProjectState, translate));
-                        return new CompilationTracker(newProject, new InProgressState(declarationOnlyCompilation, state.GeneratedDocuments, state.GeneratorDriver, compilationWithGeneratedDocuments: state.FinalCompilationWithGeneratedDocuments?.GetValueOrNull(cancellationToken), intermediateProjects));
-                    }
-
-                    return new CompilationTracker(newProject, new LightDeclarationState(declarationOnlyCompilation, state.GeneratedDocuments, state.GeneratorDriver, generatedDocumentsAreFinal: false));
-                }
```

Comment:
this case i'm still a bit unsure about @jasonmalinowski .  i could make InProgressState allow for a null compilation, and i could bring back LightDeclarationState.  But i'm not certain if they're needed.

Reply:
I'd either: [a] call it EmptyState, and allow the construction of empty states that have generated files, or [b] create a new state and call it something like "HasGeneratedFilesButNoCompilation" to make the state explicit of what's happening here. Entirely a question of how "empty" you think "empty" should be. I guess or rename Empty to NoCompilation.

Reply:
My only hesitation about making InProgressState allow nulls is then all the existing code needs to be updated; I suspect it's clearer to leave that be the "we have a compilation, but it's stale" vs. "we don't have one at all". I think in the past some of the confusion of this code is when we try making states do too much because then you don't know what they mean at all anymore.

Reply:
yeah, i went the 'NoCompilatState' route.